src/cli.cr

```diff
@@ -40,6 +270,149 @@ CLI.run(
+        alias:     "u",
+        info:      "update or delete users",
+        variables: {
+          match_name: {
+            info: "the name of the users to filter by",
+          },
+          match_groups: {
+            info: "the groups that selected users must be a member of",
+          },
+          api_key: {
+            info: "the api key to select a user by. To change a user's \
+                   API key that you already know, use the rekey command",
+          },
+        },
+        commands: {
+          edit: {
+            info:      "edit one or more user's groups",
+            action:    "DppmRestApi::CLI.edit_users",
+            alias:     "e",
+            variables: {
+              new_name:   {info: "the name to change the selected user's name to"},
+              add_groups: {
+                info: "comma-separated groups to add to the selected users",
+              },
+              remove_groups: {
+                info: "comma-separated groups to remove from the selected \
+                       users. Groups that a selected user is already not a \
+                       member of will be ignored",
+              },
+            },
+          },
+          rekey: {
+            info:      "generate a new API key for this user/users and output that file",
+            action:    "DppmRestApi::CLI.rekey_users",
+            variables: {
+              output_file: {
+                info: "write the generated key to a file. Default is to print \
+                       to stdout",
+              },
+            },
+          },
+          delete: {
+            info:   "Completely delete a user or set of users",
+            action: "DppmRestApi::CLI.delete_users",
+            alias:  "rm",
+          },
+          show: {
+            info:   "Show the available information about a user",
+            action: "DppmRestApi::CLI.show_users",
+          },
+        },
+      },
+      group: {
+        alias:    "g",
+        info:     "add, update, or delete groups",
+        commands: {
+          add: {
+            info:      "add a group",
+            alias:     "a",
+            action:    "DppmRestApi::CLI.add_group",
+            variables: {
+              id: {
+                info: "this group's ID. Defaults to a random value",
+              },
+              name: {
+                info: "a short description of this group",
+              },
+              permissions: {
+                info: "\
+                  JSON-formatted permissions, mapping a path to the query \
+                  parameters and type of access allowed on that path",
+                default: {
+                  "/**"                       => DppmRestApi::Config::Route.new(DppmRestApi::Access.deny),
+                  "/{app,pkg,src,service}/**" => DppmRestApi::Config::Route.new(
+                    DppmRestApi::Access::All,
+                    {"namespace" => ["default-namespace"]}),
+                }.to_json,
```

Comment:
That's weird. Can't it be a constant somewhere, I suppose it can be reusable it other places?

Reply:
Fair point